src/modlunky2/ui/levels/tab.py

```diff
@@ -3041,6 +3041,54 @@ def save_level(
                 if tilecode[0].split(" ", 1)[0] == "floor_hard":
                     hard_floor_code = tilecode[0].split(" ", 1)[1]
 
+            def write_vanilla_room(x, y, foreground, background):
```

Comment:
I find it really hard to reason about what's a parameter vs something from enclosing function. I'd prefer either

* Make this a separate method (needs `self`, `hard_floor_code`, and `level` templates) -- my first pick by far
* Remove all params and just use vars from outer function -- the "hmmm, you seem to enjoy chaos" option IMO, but it's shorter

The latter works because Python variables are associated at the function level, e.g. they're not specific to a for-loop.

```python
def foo(x):      
  def bar():
    print(f"bar {x*y}")
  
  for y in [7, 10]:
    bar()
  print(f"foo {y}")

foo(3)
```
Outputs
```
bar 21
bar 30
foo 10
```

Reply:
I added the parameters, but didn't add self since I want it to be an internal function still and I'm not sure if that makes sense to have a self.